### ISSUE TITLE
[Linux] Graphics like charts not working

### ISSUE THREAD

**Issue by MartinX3** [moneymanagerex/moneymanagerex#4269]
MMEX version:
 - [x] 1.5.13
 - [ ] 1.6.x

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [ ] Windows
 - [ ] Mac OSX
 - [x] Linux 

# Description of the bug

I only the an empty, white space.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Open the reports

Expected result:
My charts

Actual result:
White area

# Additional information

Everytime I open a report with charts
`** (mmex:21069): CRITICAL **: 19:17:11.814: Cannot register URI scheme memory more than once`

Specs:
```
Version: 1.5.13 64-bit
• Aufbauend auf Mär 9 2022 08:52:13
• Datenbankversion: 7 • (aes128cbc)
• Git commit: d019150be (2022-02-06)

MMEX verwendet folgene Unterstützungsprodukte:
• wxWidgets 3.1.5 (wxGTK 3.24)
• wxSQLite3 4.6.0 (SQLite 3.31.1)
• RapidJSON 1.1.0
• Lua 5.4.4
• lunasvg v2.0.1
• libcurl/7.82.0 OpenSSL/1.1.1n zlib/1.2.12 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.10.0 nghttp2/1.47.0
• gettext 0.21
• apexcharts.js

Genutztes Build:
• CMake 3.22.3
• GNU Make 4.3
• GCC 11.2.0 • EndeavourOS rolling 'rolling'

Läuft auf:
• EndeavourOS Linux "rolling"
• Linux 5.17.1-zen1-1-zen x86_64
• German (UTF-8)
• 1920x1080 24bit 96x96ppi
```

**Comment by PMaff** [comment]
Reports -> Cash flow -> Daily 
shows this already when starting version 1.5.13 on commandline with  /usr/bin/mmex on Ubuntu:
[...]
** (mmex:12723): CRITICAL **: 20:44:24.501: Cannot register URI scheme memory more than once
20:44:24: Debug: futureIgnored: 0
20:44:24: Debug: ======= mmReportCashFlow:getHTMLText =======
20:44:24: Debug: <!DOCTYPE html>
<html><head>
[...]
but the chart is shown.

**Comment by MartinX3** [comment]
Is there a way to raise the debugging level without compiling the application again?
Maybe the real reason gets filtered.

**Comment by PMaff** [comment]
I do only see the opposite with -s  :
```
$ /usr/bin/mmex --version
Unknown long option 'version'
Usage: mmex [-h] [-s] [-i <str>] [database file]
  -h, --help         	
  -s, --silent       	Do not show warning messages at startup.
  -i, --mmexini=<str>	where <str> is a path to mmexini.db3

To open a determined database(.mmb) file from a shortcut or command line, set the path to the database file as a parameter.

The file with the application settings mmexini.db3 can be used separately. Otherwise, it is taken from the home directory or the root folder of the application.
```
but with `/usr/bin/mmex -s`
there are still a lot of messages.

Btw. @whalley : the classical --version seems to be missing:
> gftp --version
gFTP 2.0.19

;-)

I do not see, what is set in mmex.
 :
https://stackoverflow.com/questions/35792613/change-wxwidgets-loglevel

**Comment by PMaff** [comment]
Tried 1.5.14-Beta.1, same here:

```
** (mmex:31766): CRITICAL **: 21:50:16.849: **Cannot register URI scheme memory more than once**
21:50:16: Debug: futureIgnored: 0
21:50:16: Debug: ======= mmReportCashFlow:getHTMLText =======
21:50:16: Debug: <!DOCTYPE html>
<html><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
<title>Money Manager Ex - Report</title>
<link href = 'memory:master.css' rel = 'stylesheet' />
<script>
    window.Promise || document.write('<script src="memory:polyfill.min.js"><\/script>');
    window.Promise || document.write('<script src="memory:classlist.min.js"><\/script>');
    window.Promise || document.write('<script src="memory:resize-observer.js"><\/script>');
    window.Promise || document.write('<script src="memory:findindex.min.js"><\/script>');
    window.Promise || document.write('<script src="memory:umd.min.js"><\/script>');
</script>
<script src = 'memory:apexcharts.min.js'></script>
<script src = 'memory:sorttable.js'></script>
<script src = 'memory:jquery.min.js'></script>
<style>
    /* Sortable tables */
    table.sortable thead {cursor: default;}
    body { font-size: 100%; }

</style>
</head>
<body>
<div class='shadowTitle'>
<header><h2>Geldfluss-Prognosen 1 Jahre im Voraus</h2></header><aside><TMPL_VAR DATE_HEADING><h5>Währung: EUR</h5><h5>Bericht erstellt 30.03.2022 21:50:16</h5></aside><footer>Konten: Alle Konten</footer></div>
<div class='shadowGraph'>
<div id='apex590186190' class='line'></div>
<script>
var chart_apex590186190 = [  ]; var options = { chart: { animations: { enabled: false }, type: 'line',  foreColor: '#373D3F', toolbar: { tools: { download: false } }, width: '95%' }, title: { text: ''}, tooltip: { theme: 'dark'  }
, colors: ['#008FFB','#00E396','#FEB019','#FF4560','#775DD0','#3F51B5','#03A9F4','#4CAF50','#F9CE1D','#FF9800','#33B2DF','#546E7A','#D4526E','#13D8AA','#A5978B','#4ECDC4','#81D4FA','#546E7A','#FD6A6A','#2B908F','#F9A3A4','#90EE7E','#FA4443','#69D2E7','#449DD1','#F86624'], xaxis: { type: 'datetime', categories: ['2022-03-30','2022-03-31','2022-04-01','2022-04-02','2022-04-03','2022-04-04','2022-04-05','2022-04-06','2022-04-07','2022-04-08','2022-04-09'
...
```
Diagram and report is shown.

**Comment by vomikan** [comment]
Pay attention to the function
const wxString getVFname4print(const wxString& name, const wxString& data)
Using wxFSFile causes the program to crash on Linux.
Therefore, the report file is saved to a temporary folder.

**Comment by PMaff** [comment]
I did not change any code, but simply tested 1.5.14-Beta.1 .

@vomikan you mean 
general_report_manager.cpp : 
mmGeneralReportManager::OnRun()
: 
```
        auto data = gr.getHTMLText();
        const auto& name = getVFname4print("grm", data);
        browser_->LoadURL(name);
```
?

**Comment by vomikan** [comment]
It should not be the "memory:" prefix in a report html code for Linux.

**Comment by MartinX3** [comment]
I added my used package versions to the first post
Maybe one of my packages i too new for the charts drawing 3rd party lib.

**Comment by whalley** [comment]
> It should not be the "memory:" prefix in a report html code for Linux.

Indeed, getVFname4print, should be removing the 'memory:' prefix prior to the LoadURL() call.

The debug output however outputs the generated HTML prior to this being called:

```
wxLogDebug("======= mmReportCashFlow:getHTMLText =======");
wxLogDebug("%s", hb.getHTMLText());
```

The error however seems to be related to the:
`browser_->RegisterHandler(wxSharedPtr<wxWebViewHandler>(new wxWebViewFSHandler("memory")));`

however I can's see where in the code the second call would be.

**Comment by whalley** [comment]
> P.S. `__WXMAC__` or `__APPLE__` ?

Stick with __WXMAC__     

https://docs.wxwidgets.org/3.1/page_cppconst.html

**Comment by whalley** [comment]
> It is not clear why the constitution does not work out: https://github.com/moneymanagerex/moneymanagerex/blob/master/src/util.cpp#L1452
> 
> I suggest changing it to
> 
> ```
> #if defined(__WXMSW__) || defined(__APPLE__)
> #else
> #endif
> ```

Sorry, just re-read your comment. What makes you think that the current `#ifndef __WXGTK__` is not working? It seems fine as is.

I thought we were asserting that the
`browser_->RegisterHandler(wxSharedPtr<wxWebViewHandler>(new wxWebViewFSHandler("memory")));`
was perhaps the nature of the issue?

**Comment by PMaff** [comment]
Just tried on Suse Tumbleweed wth latest trunk version: reports and diagrams are gone now :-(

**Comment by MartinX3** [comment]
I'm a trend setter reporting bugs by using the newest libs before all others get these bugs. :-D

**Comment by PMaff** [comment]
That is "fun" (NOT): reports and diagrams are working on Ubuntu with 
```
-- MMEX configuration summary
-- ==========================
-- Version        : 1.5.14-Beta.1
-- Commit         : cc4ad5cc2 (2022-04-02)
-- Branch         : master
-- Host system    : Linux x86_64
-- Target system  : Linux x86_64
-- Build types    : 
-- Generator      : Unix Makefiles
-- Install prefix : /usr/local
-- DB encryption  : ON
-- 
-- Versions
-- --========--
-- Linux 4.15.0-175-generic
-- Ubuntu 18.04 'bionic'
-- CMake 3.22.0
-- GNU Make 4.1
-- ccache 3.4.1
-- GNU 7.5.0
-- wxWidgets 3.1.5
-- wxSQLite3 4.6.0
-- Lua 5.3.3
-- curl 7.58.0
-- gettext 0.19.8.1
-- ------------------ 8< -----------------
-- Configuring done
-- Generating done

```
but reports and diagrams are not working on
```
-- MMEX configuration summary
-- ==========================
-- Version        : 1.5.14-Beta.1
-- Commit         : cc4ad5cc2 (2022-04-02)
-- Branch         : master
-- Host system    : Linux x86_64
-- Target system  : Linux x86_64
-- Build types    : 
-- Generator      : Unix Makefiles
-- Install prefix : /usr/local
-- DB encryption  : ON
-- 
-- Versions
-- --========--
-- Linux 5.17.1-1-default
-- openSUSE 20220331
-- CMake 3.22.3
-- GNU Make 4.3
-- ccache
-- GNU 11.2.1
-- wxWidgets 3.1.5
-- wxSQLite3 4.6.0
-- Lua 5.4.4
-- curl 7.82.0
-- gettext 0.21
-- ------------------ 8< -----------------
-- Configuring done
-- Generating done
```

**Comment by MartinX3** [comment]
Is Lua used for the diagrams?

**Comment by vomikan** [comment]
> Is Lua used for the diagrams?

No

**Comment by vomikan** [comment]
When you run any report on Linux host there should be created /tmp/mmex_user_tmp/rep.htm file.
You could investigate this file. Possibly the data for reports is not in C format (like 1,12 but not 1.12)

**Comment by PMaff** [comment]
> When you run any report on Linux host there should be created /tmp/mmex_user_tmp/rep.htm file. You could investigate this file. Possibly the data for reports is not in C format (like 1,12 but not 1.12)
> 
> One more option. Execute from a command line: LANG=C && mmex

```
"var chart_apex1559685266 = [ -23.97,-1.00,-96.66,-121.04,-10.50,-47.37,-384.07,-143.57,-68.65,-523.54,-32.00,-956.20,-42.50,-0.01,-83
.25,-61.92,-10.75,-52.22,-305.94,-979.08,-16.98,-71.10,-82.00,-7.74,-70.92,-87.50,-546.00,-49.49,-51.40,-7.50,-59.90,-0.80,-2.00,-62.
00,-19.75,-64.00,-29.21,-1399.00,-37.01,..."
```
in the /tmp file looks ok.
LANG=C && ./_CPack_Packages/Linux/RPM/mmex-1.5.14-Beta.1-Linux/usr/bin/mmex  
does not help.
Report and diagram is still not showing.

**Comment by vomikan** [comment]
Could you open the report file from /tmp folder in a web (Chrome or Mozilla) browser?

**Comment by PMaff** [comment]
I just tried with:
```
-- MMEX configuration summary
-- ==========================
-- Version        : 1.5.14-Beta.1
-- Commit         : cc4ad5cc2 (2022-04-02)
-- Branch         : master
-- Host system    : Linux x86_64
-- Target system  : Linux x86_64
-- Build types    : 
-- Generator      : Unix Makefiles
-- Install prefix : /usr/local
-- DB encryption  : ON
-- 
-- Versions
-- --========--
-- Linux 5.17.1-1-default
-- openSUSE 20220403
-- CMake 3.23.0
-- GNU Make 4.3
-- ccache
-- GNU 11.2.1
-- wxWidgets 3.1.5
-- wxSQLite3 4.6.0
-- Lua 5.4.4
-- curl 7.82.0
-- gettext 0.21
```

Can view the report 
/tmp/mmex_pmaff_tmp/rep.html
with chromium.

**Comment by PMaff** [comment]
> This means that the report is formed absolutely correctly. But the graphs are not displayed in the application on the openSUSE platform?

I see that in the same way. 
Graphs and reports are missing, white screen in MMEX only.

Looks like the same for ARCH Linux that @MartinX3 uses.

**Comment by PMaff** [comment]
> Does the WebView sample work? https://github.com/wxWidgets/wxWidgets/tree/v3.1.5/samples/webview

.../wxWidgets-3.1.5/gtk-build/samples/webview/webview
seems to work without problems.

**Comment by whalley** [comment]
This may be a dumb question... but does the HTML view on the Home Page display OK?

**Comment by PMaff** [comment]
> This may be a dumb question... but does the HTML view on the Home Page display OK?

You mean displaying the mmex homepage with wxWidget-webview test?

**Comment by MartinX3** [comment]
> This may be a dumb question... but does the HTML view on the Home Page display OK?

At my side I only see a white area.
![image](https://user-images.githubusercontent.com/4518092/162616325-87d07704-6fb7-4c2d-9cd8-c7530ab309d5.png)

**Comment by PMaff** [comment]
> This may be a dumb question... but does the HTML view on the Home Page display OK?

If you mean the mmex homepage, then webview test shows it like this on Suse:
![wxWidgetwebview_mmexhompage](https://user-images.githubusercontent.com/93879362/162617057-d2ccb49d-9d7a-45cd-b630-2a43018f2549.jpg)

**Comment by PMaff** [comment]
> This may be a dumb question... but does the HTML view on the Home Page display OK?

I do not know if it helps you but the "update" Window on version 1.5.13 on Suse with
```
Money Manager Ex
Version: 1.5.13 64-bit
• Aufbauend auf Feb 10 2022 03:16:33
• Datenbankversion: 7 • (aes128cbc)
• Git commit: d019150be (2022-02-06)
MMEX verwendet folgene Unterstützungsprodukte:
• wxWidgets 3.1.5 (wxGTK 3.24)
• wxSQLite3 4.6.0 (SQLite 3.31.1)
• RapidJSON 1.1.0
• Lua 5.4.4
• lunasvg v2.0.1
• libcurl/7.82.0 OpenSSL/1.1.1n-fips zlib/1.2.11 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libpsl/0.21.1 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.47.0 OpenLDAP/2.5.9
• gettext 0.21
• apexcharts.js
Genutztes Build:
• CMake 3.22.2
• GNU Make 4.3
• GCC 11.2.1 20220103 [revision d4a1d3c4b377f1d4acb34fe1b55b5088a3f293f6] • openSUSE 20220207
Läuft auf:
• openSUSE Tumbleweed "n/a"
• Linux 5.17.1-1-default x86_64
• English (UTF-8)
• 0x0 0bit 96x96ppi
```
also is not displayed:
![mmex_updatepage1_5_13](https://user-images.githubusercontent.com/93879362/162617349-814bb0ea-e0af-4706-84a7-a02241638c15.jpg)

**Comment by MartinX3** [comment]
Same on arch linux. Update window is empty.

**Comment by whalley** [comment]
> > This may be a dumb question... but does the HTML view on the Home Page display OK?
> 
> At my side I only see a white area. ![image](https://user-images.githubusercontent.com/4518092/162616325-87d07704-6fb7-4c2d-9cd8-c7530ab309d5.png)

Thanks, so this confirms that the WebView is broken in those builds for some odd reason.

> reports and diagrams are working on Ubuntu
moneymanagerex/build$ ../_CPack_Packages/Linux/DEB/mmex-1.5.14-Beta.1-Linux/usr/bin/mmex

I can see @PMaff  reports that it as working in
• Ubuntu 18.04.6 LTS "bionic" with wxWidgets 3.1.5 (wxGTK 3.22)

But other Linux builds using • wxWidgets 3.1.5 (wxGTK 3.24) are reporting as not working.

Is it possible to test one of the 'non-workiong builds with wxGTK 3.22 ?

**Comment by PMaff** [comment]
I am not sure, if this makes sense since the webview test application is working correct.
If it would be a problem of wxWidget 3.1.5 + wxGTK 3.24 then webview should not display anything -> ?

**Comment by PMaff** [comment]
Just for fun I compiled wxWidget 3.1.7 on Suse Tumbleweed.
Compilation of mmex afterwards was ok
```
Version: 1.5.14-Beta.1 64-bit
• Build on Apr 10 2022 22:37:13
• Database version: 7 • (aes128cbc)
• Git commit: cc4ad5cc2 (2022-04-02)
• Git branch: master
MMEX is using the following support products:
• wxWidgets 3.1.7 (wxGTK 3.24)
• wxSQLite3 4.6.0 (SQLite 3.31.1)
• RapidJSON 1.1.0
• Lua 5.4.4
• lunasvg v2.0.1
• libcurl/7.82.0 OpenSSL/1.1.1n-fips zlib/1.2.11 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libpsl/0.21.1 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.47.0 OpenLDAP/2.5.9
• gettext 0.21
• apexcharts.js
Build using:
• CMake 3.23.0
• GNU Make 4.3
• GCC 11.2.1 20220316 [revision 6a1150d1524aeda3381b2171712e1a6611d441d6] • openSUSE 20220408
Running on:
• openSUSE Tumbleweed "n/a"
• Linux 5.17.1-1-default x86_64
• English (United Kingdom) (UTF-8)
• 0x0 0bit 96x96ppi
```

Same problem with reports and graphs.

**Comment by moritzhoewer** [comment]
Hi,
I am running ArchLinux and just updated to mmex 1.5.14 and wxWidgets 3.1.6. That broke the WebView for me as well. (White page like @MartinX3 reported in https://github.com/moneymanagerex/moneymanagerex/issues/4269#issuecomment-1094252550)

Opening the HTML in a Webbrowser (Firefox in my case) works and displays everything, including graphs.

**Comment by moritzhoewer** [comment]
I can also confirm what @PMaff said in https://github.com/moneymanagerex/moneymanagerex/issues/4269#issuecomment-1086869479 - downgrading wxWidget to 3.1.5 and/or downgrading mmex to 1.5.13 does not help. WebView is still broken in all combinations.

And same as https://github.com/moneymanagerex/moneymanagerex/issues/4269#issuecomment-1090180960, the sample from wxWidgets works flawlessly.

And I'm not sure if it's related but I also see the
```
** (mmex:118094): CRITICAL **: 19:25:20.349: Cannot register URI scheme memory more than once
```
every time I switch to a new report or back to the homepage

**Comment by whalley** [comment]
Finally got around to building on the Linux to investigate this and I can't see the issue?

Built on Ubuntu 10.04 ARM64, see below.

![CleanShot 2022-04-15 at 19 45 24](https://user-images.githubusercontent.com/487345/163609086-6f14beb0-3709-43b2-a5d1-b51c4863c4bd.png)

Installed 

wxWidgets build

```
sudo apt-get install -y libwebkit2gtk-4.0-dev
cd ../../wx/build/
$ ../configure --disable-shared --enable-cxx11 --with-cxx=11 --without-libtiff --enable-webview --with-gtk=3 --enable-unicode --enable-intl
$ make
```

mmex build / install

```
$ cmake -DwxWidgets_CONFIG_EXECUTABLE=/home/mark/GitRepos/wxWidgets/build-ubuntu/wx-config -DCMAKE_BUILD_TYPE=Release ..
$ make
$ sudo make install
```

**Comment by moritzhoewer** [comment]
I stumbled upon this forum post (https://forums.wxwidgets.org/viewtopic.php?p=196101#p196101) that mentions a blank webview when unicode characters are involved.

I was able to reproduce this in the webview sample apllication by replacing the `OnSetPage` function in line 926 of `webview.cpp` with
```cxx
void WebFrame::OnSetPage(wxCommandEvent& WXUNUSED(evt))
{
    m_browser->SetPage
               (
                "<html><title>New Page</title>"
                "<body>Hello ⇨ World</body></html>",
                wxString()
               );
}
```
where the `⇨` is a unicode glyph.

This yields
![image](https://user-images.githubusercontent.com/1266361/163651503-4f2446c4-a971-4f97-88f2-1178cb6077b5.png)
when clicking on "Menu --> Set page text". The same happens with the characters from the forum post.

If I change the function to
```cxx
void WebFrame::OnSetPage(wxCommandEvent& WXUNUSED(evt))
{
    m_browser->SetPage
               (
                wxString::FromUTF8("<html><title>New Page</title>"
                "<body>Hello ⇨ World</body></html>"),
                wxString()
               );
}
```
it works
![image](https://user-images.githubusercontent.com/1266361/163651584-04d17871-b01b-4cb2-979e-b9513bbfe321.png)

I looked into the mmex code, but it was my first time and the logic is a little more complicated than above, so I gave up :sweat_smile: . I did see UTF-8 conversions in many places, but maybe there is one place where one was missed? From my testing with the sample app, it seems that the UTF-8 conversion has to happen when the C++ string first turns into a `wxString`. If that gets messed up, you pretty much lost.

Maybe one of you, who is more familiar with the codebase, has an idea?

**Comment by whalley** [comment]
Ignore last commit that was for another issue (finger trouble on my part!)

**Comment by whalley** [comment]
Seems fine on my Ubuntu 20.04 ARM64 v1.5.14 build

![CleanShot 2022-04-16 at 11 09 31](https://user-images.githubusercontent.com/487345/163671009-d93d5b55-9375-4154-a6c3-92a4ac841f2c.png)

**Comment by vomikan** [comment]
It's a samle of the begin of the report:
![image](https://user-images.githubusercontent.com/6836805/163674275-5e7d34da-d869-4e90-b19a-900b86d5d270.png)

Regarding the following article:
https://www.w3schools.com/html/html_charset.asp
it should be:
`<meta charset="UTF-8">`
It's unlikely, but it might make a sence.

**Comment by moritzhoewer** [comment]
I tried setting the charset - didn't make a difference in the example. Thinking about it again though, it still doesn't really make sense that this would be unicode related. Why would it then suddenly break (also in the old version) and why does it work for @whalley.

I guess it's more likely this is some other (system) dependency - the problem there for me is that I updated around 270 packages when I broke my webview. I tried downgrading all the packages that seemed related to GTK / wxSomething / webkit but that didn't help. If I find time, I can maybe try downgrading more or all of them to try finding the one package that caused the issue.

**Comment by vomikan** [comment]
@moritzhoewer could you show 
localectl status

**Comment by MartinX3** [comment]
```bash
$ localectl status
   System Locale: LANG=de_DE.UTF-8
       VC Keymap: de-latin1-nodeadkeys
      X11 Layout: de
       X11 Model: pc105
     X11 Variant: nodeadkeys
     X11 Options: terminate:ctrl_alt_bksp
```
```bash
$ locale -a
C
de_DE.utf8
en_US.utf8
POSIX
```

**Comment by moritzhoewer** [comment]
```
➜  ~ localectl status
   System Locale: LANG=en_US.UTF-8
                  LC_TIME=de_DE.UTF-8
                  LC_PAPER=de_DE.UTF-8
       VC Keymap: de
      X11 Layout: de
```
```
➜  ~ locale -a
C
de_DE.utf8
en_US.utf8
POSIX
```

**Comment by vomikan** [comment]
Could you try
`LANG="C" && mmex`
or
`LANG="en_US.utf8" && mmex`

**Comment by MartinX3** [comment]
It doesn't fix the issue.

**Comment by moritzhoewer** [comment]
Same here

**Comment by vomikan** [comment]
Try this fix for luck.
![image](https://user-images.githubusercontent.com/6836805/163678774-983e50e0-43b7-4af6-8da5-c53b61975be9.png)

**Comment by moritzhoewer** [comment]
I tried both and it doesn't work still. But from my (limited) testing with the wxWidgets example yesterday night, this would be expected. The "corruption" occurs when data is **first added** to a `wxString` - after that, I found no way to "fix" it again.

**Comment by PMaff** [comment]
Hmm, on Ubuntu 18.04 LTS , where it is working, I get 
```
localectl status
   System Locale: LANG=en_GB.UTF-8
                  LC_NUMERIC=de_AT.UTF-8
                  LC_TIME=de_AT.UTF-8
                  LC_MONETARY=de_AT.UTF-8
                  LC_PAPER=de_AT.UTF-8
                  LC_NAME=de_AT.UTF-8
                  LC_ADDRESS=de_AT.UTF-8
                  LC_TELEPHONE=de_AT.UTF-8
                  LC_MEASUREMENT=de_AT.UTF-8
                  LC_IDENTIFICATION=de_AT.UTF-8
       VC Keymap: n/a
      X11 Layout: de
       X11 Model: pc105
```
```
locale -a
C
C.UTF-8
de_AT
de_AT@euro
de_AT.iso88591
de_AT.iso885915@euro
de_AT.utf8
en_AG
en_AG.utf8
en_AU.utf8
en_BW.utf8
en_CA.utf8
en_DK.utf8
en_GB
en_GB.iso88591
en_GB.iso885915
en_GB.utf8
en_HK.utf8
en_IE.utf8
en_IL
en_IL.utf8
en_IN
en_IN.utf8
en_NG
en_NG.utf8
en_NZ.utf8
en_PH.utf8
en_SG.utf8
en_US.utf8
en_ZA.utf8
en_ZM
en_ZM.utf8
en_ZW.utf8
POSIX
```

I never changed this so it seem to be a lot of unnecessary stuff here on my Ubuntu 18.04

**Comment by moritzhoewer** [comment]
From the comments so far, it seems to be broken for @MartinX3 and myself on ArchLinux, and for @PMaff on a recent OpenSUSE. But it's working on Ubuntu 18.04 and 20.04. (which are not as "bleeding edge")

Sounds to me like an issue with a new version of some system library

**Comment by PMaff** [comment]
Does this help:
"wx.html2.WebView with WebKit 2 does not display pictures from the disk"
https://github.com/wxWidgets/Phoenix/issues/1071
?
On Ubuntu 18.04 LTS I have the version `2.4.11-3ubuntu3` of `libwebkitgtk-dev`.
libwebkit2gtk is version 2.36.0-2.1 from Suse according to yast2.

**Comment by moritzhoewer** [comment]
> The rep.html in Download and the one in /tmp/mmex_pmaff_tmp/ are the same
but are displayed differently when called directly or via
/usr/libexec/libwebkit2gtk-4_1-0/MiniBrowser

Image looks like CSS stylesheet is missing - that one is probably referenced via relative path and can't be found in Downloads since you only downloaded the HTML. If you open it in `/tmp...`, then the CSS is also there alongside it.

**Comment by PMaff** [comment]
> > The rep.html in Download and the one in /tmp/mmex_pmaff_tmp/ are the same
> > but are displayed differently when called directly or via
> > /usr/libexec/libwebkit2gtk-4_1-0/MiniBrowser
> 
> Image looks like CSS stylesheet is missing - that one is probably referenced via relative path and can't be found in Downloads since you only downloaded the HTML. If you open it in `/tmp...`, then the CSS is also there alongside it.

..link href = 'master.css' rel = 'stylesheet' .. from rep.html  cannot be in $HOME/Downloads of course.  Silly me.
But why does the webkit test browser on Ubuntu load the report correctly and on Suse it calls the default browser?
On Ubuntu also the default browser is set as firefox.

**Comment by moritzhoewer** [comment]
> libwebkit2gtk is version 2.36.0-2.1 from Suse according to yast2.

Same here, on my ArchLinux, I have
```
[2022-04-15T16:42:03+0200] [ALPM] upgraded webkit2gtk-4.1 (2.34.6-1 -> 2.36.0-2)
[2022-04-15T16:43:42+0200] [ALPM] upgraded webkit2gtk (2.34.6-1 -> 2.36.0-2)
```
I tried downgrading back to 2.34.6-1 yesterday already. Just tried again and then also rebuilt wxWidgets and mmex. Nothing changes.


Another thing that came to my mind though, based on the suggestion from @vomikan in https://github.com/moneymanagerex/moneymanagerex/issues/4269#issuecomment-1100675637

I made the function return `wxString("Hello World")`, which results in the following file
```
➜  ~ cat /tmp/mmex_moritz_tmp/rep.html
Hello World
➜  ~ 
```
Still nothing is showing in mmex. Just a blank page. (Needless to say the file renders fine in firefox). That file contains no unicode characters and neither does its name or its path in the filesystem.

Maybe @PMaff is onto something with the webkit test browser

**Comment by PMaff** [comment]
I am just trying to compile webkitgtk-2.4.11.tar.xz on my Suse Tumbleweed test machine.

**Comment by moritzhoewer** [comment]
I went back to the wxWidgets WebView sample - @PMaff already mentioned in https://github.com/moneymanagerex/moneymanagerex/issues/4269#issuecomment-1090180960 that the sample refuses to display the page (actually any file:// URL I tried).

![image](https://user-images.githubusercontent.com/1266361/163691069-2f62031d-2128-4d31-a09c-9e7139c7c59f.png)

The indicated `wxWEBVIEW_NAV_ERR_OTHER` seems to be the default error when the real cause can't be found (see https://github.com/wxWidgets/wxWidgets/search?q=wxWEBVIEW_NAV_ERR_OTHER)

The underlying function both in mmex and in the sample is `wxWebView::LoadURL()`, so this might be a wxWidgets bug after all.

**Comment by moritzhoewer** [comment]
Should I create an issue there?

**Comment by PMaff** [comment]
webkitgtk-2.4.11.tar.xz does not compile with error:
make[1]: *** [GNUmakefile:70161: Source/JavaScriptCore/API/libjavascriptcoregtk_3_0_la-JSStringRef.lo] Error 1
:-(

**Comment by moritzhoewer** [comment]
### I found the problem
and it's nothing we considered so far. Seems to be related to the mime-type database.

After executing
```
rm ~/.local/share/mime/packages/user-extension-html.xml
update-mime-database ~/.local/share/mime
```
as per https://stackoverflow.com/questions/71734719/webkitgtk-doesnt-load-local-files/71740957#71740957, the WebView sample can display local files again and so can mmex. (@PMaff this also explains your Download issue under openSUSE if I read the StackOverflow post correctly)

**Comment by PMaff** [comment]
> ### I found the problem
> 
> and it's nothing we considered so far. Seems to be related to the mime-type database.
> 
> After executing
> 
> ```
> rm ~/.local/share/mime/packages/user-extension-html.xml
> update-mime-database ~/.local/share/mime
> ```
> 
> as per https://stackoverflow.com/questions/71734719/webkitgtk-doesnt-load-local-files/71740957#71740957, the WebView sample can display local files again and so can mmex. (@PMaff this also explains your Download issue under openSUSE if I read the StackOverflow post correctly)

That's it.
On Suse Tumbleweed reports and graph is working again for 1.5.13 stable and 1.5.14-Beta.1 !
Cudos to @moritzhoewer ;-)

**Comment by PMaff** [comment]
@MartinX3 if this also works for you, then I guess you can close this ticket. ;-)

**Comment by MartinX3** [comment]
Weird.
`update-mime-database ~/.local/share/mime` fixed the issue.

I don't need to remove `~/.local/share/mime/packages/user-extension-html.xml`.
So it's only the `update-mime-database` command.

I wonder if I need to execute it again on the next wxWidgets or mmex update.

Thank you.

**Comment by mrjavum** [comment]
I have tried on Arch
`update-mime-database ~/.local/share/mime`
and the problem not fixed.
But 
`rm ~/.local/share/mime/packages/user-extension-html.xml
update-mime-database ~/.local/share/mime`
fixed the problem

**Comment by MartinX3** [comment]
@mrjavum 
Oh, then I should remove the file.

I renamed it, updated the database and named it back.
So this file makes trouble.
I wonder what created it and why it got created.

**Comment by PMaff** [comment]
> ### I found the problem
> and it's nothing we considered so far. Seems to be related to the mime-type database.
> 
> After executing
> 
> ```
> rm ~/.local/share/mime/packages/user-extension-html.xml
> update-mime-database ~/.local/share/mime
> ```
> 
> as per https://stackoverflow.com/questions/71734719/webkitgtk-doesnt-load-local-files/71740957#71740957, the WebView sample can display local files again and so can mmex. (@PMaff this also explains your Download issue under openSUSE if I read the StackOverflow post correctly)

On Suse Tumbleweed this happens very often after doing a zypper dup.

**Comment by kennydrobnack** [comment]
> ### I found the problem
> and it's nothing we considered so far. Seems to be related to the mime-type database.
> 
> After executing
> 
> ```
> rm ~/.local/share/mime/packages/user-extension-html.xml
> update-mime-database ~/.local/share/mime
> ```
> 
> as per https://stackoverflow.com/questions/71734719/webkitgtk-doesnt-load-local-files/71740957#71740957, the WebView sample can display local files again and so can mmex. (@PMaff this also explains your Download issue under openSUSE if I read the StackOverflow post correctly)

I just want to say THANK YOU for finding this fix. I had been wanting to run MMEX on Pop!_OS 22.04 LTS release and it was bugging the crap out of me that all reports I tried to run were showing up as blank and resulting in the error `Cannot register URI scheme memory more than once`

This was a pretty stock install with almost no customizations and I was so excited that this was working without having to build from source.
